lib/types/HttpStream.ts

```diff
@@ -42,5 +43,45 @@ export class HttpStream {
   ) {
     this.stream = readableStream;
     this.totalBytes = totalBytes;
+    this.destroyed = readableStream.destroyed;
+  }
+
+  /**
+   * Returns if the stream is errored
+   */
+  isErrored (): boolean {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
I think we should disable this eslint rule since it's prevent use to disable typescript when needed and we end up with 2 lines of comments